src/h3lib/lib/polygon.c

```diff
@@ -93,3 +95,172 @@ bool pointInsidePolygon(const GeoPolygon* geoPolygon, const BBox* bboxes,
 #undef IS_EMPTY
 #undef INIT_ITERATION
 #undef ITERATE
+
+/**
+ * Count the number of polygons containing a given loop.
+ * TODO: It would be really nice to memoize this.
+ * @param  loop         Loop to count containers for
+ * @param  polygons     Polygons to test
+ * @param  polygonCount Number of polygons in the test array
+ * @return              Number of polygons containing the loop
+ */
+static int countContainers(LinkedGeoLoop* loop, LinkedGeoPolygon** polygons,
+                           BBox** bboxes, int polygonCount) {
+    int containerCount = 0;
+    for (int i = 0; i < polygonCount; i++) {
+        if (pointInsideLinkedGeoLoop(polygons[i]->first, bboxes[i],
+                                     &loop->first->vertex)) {
+            containerCount++;
+        }
+    }
+    return containerCount;
+}
+
+/**
+ * Find the polygon to which a given hole should be allocated. Note that this
+ * function will return null if no parent is found.
+ * @param  loop         Inner loop describing a hole
+ * @param  polygon      Head of a linked list of polygons to check
+ * @param  polygonCount Number of polygons to check
+ * @return              Pointer to parent polygon, or null if not found
+ */
+static LinkedGeoPolygon* findPolygonForHole(LinkedGeoLoop* loop,
+                                            LinkedGeoPolygon* polygon,
+                                            BBox* bboxes, int polygonCount) {
```

Comment:
`bboxes` are not part of above `@param` documentation, making the logic here harder to follow.

Reply:
Will add to the docs. A `bbox` is necessary for each polygon in order to run the `pointInside` algorithm, which relies on the `bbox` for fast failure and identification of transmeridian polygons.